### PR TITLE
Fix variable defaulting resolution bug/ remove deprecated overridestring/ enhance testing

### PR DESCRIPTION
**Pull request by davidgamero**
# Description

Fixes a current bug where variable defaults are not resolved properly for references

Added unit tests for prompting logic and default resolution to avoid this issue in the future

Refactor prompt package to allow better testing by adding parameters for Stdin and Stdout with defaults on nil to follow `promptui` library convention

Remove deprecated overridestring structs

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [x]  Existing tests pass
- [x] Added unit tests for string prompting
- [x]  Added unit tests for variable defaulting resolution

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by tbarnes94 — APPROVED** [review]

File: pkg/prompts/prompts.go [hunk 1]
```diff
@@ -31,66 +27,27 @@ func RunPromptsFromConfigWithSkips(config *config.DraftConfig, varsToSkip []stri
 	inputs := make(map[string]string)
 
 	for _, customPrompt := range config.Variables {
+		promptVariableName := customPrompt.Name
 		if _, ok := skipMap[customPrompt.Name]; ok {
```
Comment (OliverMKing) [inline pkg/prompts/prompts.go:31]: should use promptVariableName as the key here.

**Review by OliverMKing — APPROVED** [review]
Makes sense to me.

File: pkg/prompts/prompts.go (hunk 1 above)
Comment (davidgamero) [reply pkg/prompts/prompts.go:31]: good catch thanks

**Review by OliverMKing — APPROVED** [review]

File: pkg/prompts/prompts.go [hunk 2]
```diff
@@ -104,6 +61,71 @@ func RunPromptsFromConfigWithSkips(config *config.DraftConfig, varsToSkip []stri
...
+
+	_, input, err := newSelect.Run()
+	if err != nil {
+		return "", err
+	}
+	return input, nil
+}
+
+// RunStringPrompt runs a prompt for a string variable, returning the user string input for the prompt
+func RunStringPrompt(customPrompt config.BuilderVar, defaultValue string, Stdin io.ReadCloser, Stdout io.WriteCloser) (string, error) {
```
[the remark this answers is not in the document]
Comment (davidgamero) [reply pkg/prompts/prompts.go]: great point- just added as a parameter with some sane defaults on nil validator

**Review by OliverMKing — APPROVED** [review]

**Pull request merged**